src/components/renderer/renderer.tsx

```diff
@@ -57,37 +57,36 @@ class Editor extends React.PureComponent<Props, State> {
     }
   }
 
-  public isResponsive(): [boolean, boolean] {
+  public isResponsive() {
     const spec = this.props.vegaSpec;
-    let w = false;
-    let h = false;
-    if (spec.signals && spec.signals instanceof Array) {
+    let responsiveWidth = false;
+    let responsiveHeight = false;
+    if (spec.signals) {
       for (const signal of spec.signals) {
-        if (signal.name == 'width' && (signal as vega.InitSignal).init == 'containerSize()[0]') {
-          w = true;
+        if (signal.name == 'width' && (signal as vega.InitSignal).init.indexOf('containerSize') >= 0) {
+          responsiveWidth = true;
         }
-        if (signal.name == 'height' && (signal as vega.InitSignal).init == 'containerSize()[1]') {
-          h = true;
+        if (signal.name == 'height' && (signal as vega.InitSignal).init.indexOf('containerSize') >= 0) {
+          responsiveHeight = true;
         }
       }
     }
-    return [w, h];
+    return {responsiveWidth, responsiveHeight};
   }
 
   public handleResizeMouseDown(e: React.MouseEvent) {
     const x0 = e.pageX;
     const y0 = e.pageY;
-    const width0 = this.state.width;
-    const height0 = this.state.height;
-    const [resizeWidth, resizeHeight] = this.isResponsive();
+    const {width: width0, height: height0} = this.state;
```

Comment:
Do you want to leave this as it was or also change the `x0`, `y0` above and `x1`, `y1` below?

Reply:
I'd prefer not to rename in destructor. Reverted back, also added a few more comments to the code. I think it's good to go. Do you prefer squash merge or regular merge?